# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
****

In [12]:
import pandas as pd
import numpy as np
from scipy import stats

In [13]:
data = pd.io.stata.read_stata('us_job_market_discrimination.dta')

In [14]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [15]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

Question 1. The appropriate test for this scenario is Hypothesis Test with Binomial distribution. CLT applies because each event is the distribution is the sum of mutually independent events.

Question 2. 
The null hypothesis: callback rate for a black-sounding name = callback rate for a white-sounding name
Alternate hypothesis: callback rate for a black-sounding name < callback rate for a white-sounding name

In [31]:

# of black-sounding resumes sent
sample_size = len(data[data.race=='b'])

# of white-sounding resumes sent
total_w_size = len(data[data.race=='w'])


#callback rates for black and white-sounding names
callback_b = sum(data[data.race=='b'].call)
callback_w = sum(data[data.race=='w'].call)

#total callbacks and resumes sent
total_sent = sample_size + total_w_size
total_callbacks = callback_b + callback_w

#total population callback rate
p_total_proportion = total_callbacks/total_sent

#probability of callback for white-sounding name
p_w_proportion = callback_w/total_w_size

#Find probability of callback for black-sounding name
p_proportion = callback_b/sample_size
print('Callback rate for black-sounding names: ' + str(p_proportion))

#Margin of Error (95% and 99% Confidence) for black-sounding names
ME_95 = 1.96 * ((p_proportion*(1-p_proportion)/sample_size)**(0.5))
ME_99 = 2.57 * ((p_proportion*(1-p_proportion)/sample_size)**(0.5))

print('Margin of error for 95% CI is: ' + str(ME_95))
print('Margin of error for 95% CI is: ' + str(ME_99))

#Confidence Intervals, 95% and 99%
CI_95 = [p_proportion - ME_95, p_proportion + ME_95]
CI_99 = [p_proportion - ME_99, p_proportion + ME_99]

print('95% Confidence Interval is: ' + str(CI_95))
print('99% Confidence Interval is: ' + str(CI_99))

#Find Z
Z = (p_proportion - p_w_proportion)/np.sqrt(p_total_proportion*(1-p_total_proportion)*((1/sample_size) + (1/total_w_size)))

print('Z: ' + str(Z))
print('P-Value: ' + str(stats.norm.cdf(Z)))

Callback rate for black-sounding names: 0.064476386037
Margin of error for 95% CI is: 0.00975515802791
Margin of error for 95% CI is: 0.012791202108
95% Confidence Interval is: [0.05472122800904957, 0.074231544064872396]
99% Confidence Interval is: [0.051685183928934286, 0.077267588144987687]
Z: -4.10841215243
1.99194341879e-05


Question 3. 
Callback rate for black-sounding names: 0.064476386037

Margin of error for 95% CI is: 0.00975515802791

Margin of error for 95% CI is: 0.012791202108

95% Confidence Interval is: [0.05472122800904957, 0.074231544064872396]

99% Confidence Interval is: [0.051685183928934286, 0.077267588144987687]

P-Value = P(Z < -4.108) = 1.99194341879e-05

Question 4.
In the context of the problem, we reject the null hypothesis because we have a statistically significant hypothesis test that shows that the rate of callback for resumes with black-sounding names is less than that of resumes with white-sounding names. However, we don't necessarily claim that discrimination exists, because just because there is a statistically significant difference, we aren't sure what exactly is the cause quite yet.

Question 5.

The analysis does not necessarily mean that race/name is the biggest factor in getting callbacks, because the data has many more dimensions that could potentially be the true underlying variables as to the difference in callback rates. 

Other very important variables in job callback rates also include years of experience, number of jobs, and years of education. Solely running a hypothesis test across one variable without giving considerations to others would be incomplete in making any strong conclusion or accusation such as racial discrimination.